Zend/zend_vm_execute.h

```diff
@@ -1779,6 +1772,27 @@ static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_HANDLE_EXCEPTION_SPEC_HANDLER(
 
 	cleanup_unfinished_calls(execute_data, throw_op_num);
 
+	if (throw_op->result_type & (IS_VAR | IS_TMP_VAR)) {
+		switch (throw_op->opcode) {
+			case ZEND_ADD_ARRAY_ELEMENT:
+			case ZEND_ROPE_ADD:
+				break; /* exception while building structures, live range handling will free those */
+
+			case ZEND_FETCH_CLASS:
+			case ZEND_DECLARE_CLASS:
+			case ZEND_DECLARE_INHERITED_CLASS:
+			case ZEND_DECLARE_ANON_CLASS:
+			case ZEND_DECLARE_ANON_INHERITED_CLASS:
+				break; /* return value is zend_class_entry pointer */
+
+			case ZEND_JMP_SET:
+				break; /* takes care of handling itself */
```

Comment:
Right now you still have to the explicit `EG(exception)` check -- it's just in the other branch. What I'd do here is handle JMP_SET normally and use an UNDEF BP_VAR_R fetch and handle the UNDEF CV explicitly only in the !is_true branch (this is the only case where the exception check / undef would be necessary)